packages/compiler/src/core/messages.ts

```diff
@@ -736,6 +736,7 @@ const diagnostics = {
       default: "Invalid encoding",
       wrongType: paramMessage`Encoding '${"encoding"}' cannot be used on type '${"type"}'. Expected: ${"expected"}.`,
       wrongEncodingType: paramMessage`Encoding '${"encoding"}' on type '${"type"}' is expected to be serialized as '${"expected"}' but got '${"actual"}'.`,
+      wrongUnixTimestampEncoding: paramMessage`Encoding '${"encoding"}' on type '${"type"}' is expected to be serialized as '${"expected"}' but got '${"actual"}'. Set '@encode' 2nd parameter to be an integer. e.g. '@encode("unixTimestamp", int32)'`,
```

Comment:
Wondering if we should have a more generic explaination, This was reported to be the same problem for duration with `seconds` encoding

Reply:
I can just do it for all the non-string encodings?

Reply:
yeah I think that would be great